### ISSUE TITLE
return value of web3.eth.sendTransaction

### ISSUE THREAD

**Issue by palango** [ethereum/web3.py#686]
* Version: 4.0.0-beta.11
* Python:  3.6
* OS: osx

### What was wrong?

The return value of `web3.eth.sendTransaction(...)` seems to be HexBytes, not a str as shown in the docs.

### How can it be fixed?

Fix the return value or fix the docs.

**Comment by carver** [comment]
:+1: documentation fix TODO

**Comment by lmittmann** [comment]
Why has the return value been changed to `HexBytes`? I always need the string representation.

This is one of many changes in `4.0.0` that require lots of updates in my code base and are really annoying. Now I need to mess up my code with lots of ` Web3.toHex(...)` conversions.

**Comment by pipermerriam** [comment]
@lmittmann I'm sorry that the v3 -> v4 changes are causing you problems, however, there are good reasons behind all of these changes.

The `HexBytes` one specifically is part of a broader change removing ambiguity about return values and input values.  

Given a string `"0xabcd"` we can't infer if this is the hex encoded value `b'\xab\xcd'`, or the literal string `"0xabcd"`.  When returning values like this, we chose to use a `bytes` representation as that removes the ambiguity.  However, we found that returning bytes was very *unfriendly* UX when inspecting return values, so we came up with a thin wrapper around the `bytes` type, `HexBytes` that when represented in a terminal, shows the hex encoded value.
